Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,289 @@
+                    .getConceptTypeByConceptId(selectedId)) {
+                alternativeIds.add(conceptId);
+                ConceptEntry conceptEntry = conceptManager.getConceptEntry(conceptId);
+                alternativeIds.addAll(conceptEntry.getAlternativeIds());
+            }
+        }
+        entry.setAlternativeIds(alternativeIds);
+    }
+
+    private void addMergedIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        List<String> conceptIds = conceptMergeBean.getConceptIds();
+        String selectedId = conceptMergeBean.getSelectedConceptId();
+        
+        StringBuffer mergedIdsBuffer = new StringBuffer();
+        String prefix = "";
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(selectedId) || ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                    .getConceptTypeByConceptId(selectedId)) {
+                // If its wordnet add it to mergedIds because for wordnet we
+                // will be creating a new wrapper.
+                mergedIdsBuffer.append(prefix);
+                mergedIdsBuffer.append(conceptId);
+                prefix = ",";
+            }
+        }
+        entry.setMergedIds(mergedIdsBuffer.toString());
+    }
+
+    private Set<String> getAllConceptLists() {
+        List<ConceptList> allLists = conceptListManager.getAllConceptLists();
+        Set<String> conceptLists = new HashSet<>();
+        for (ConceptList conceptList : allLists) {
+            conceptLists.add(conceptList.getConceptListName());
+        }
+        return conceptLists;
+    }
+
+    private Set createSet(String value, Set collection) {
+
+        if (isNullOrEmpty(value)) {
+            return null;
+        }
+
+        if (collection == null) {
+            collection = new HashSet();
+        }
+        collection.add(value);
+        return collection;
+    }
+
+    private ConceptType[] getTypesByTypeId(Set<String> conceptTypesId) {
+        ConceptType[] types = new ConceptType[conceptTypesId.size()];
+        int i = 0;
+        for (String typeId : conceptTypesId) {
+            types[i] = conceptTypeManager.getType(typeId);
+            i++;
+        }
+        return types;
+    }
+
+    private boolean isNullOrEmpty(String value) {
+        if (value == null) {
+            return true;
+        }
+
+        if (value.trim().isEmpty()) {
+            return true;
+        }
+        return false;
+    }
+
+    private Set<String> combineSynonymIds(Set<String> existingSynonymId, String newSynonymIds) {
+
+        if (newSynonymIds == null) {
+            return existingSynonymId;
+        }
+
+        if (newSynonymIds.trim().isEmpty()) {
+            return existingSynonymId;
+        }
+
+        String[] synonymIds = newSynonymIds.split(",");
+
+        for (String synonymId : synonymIds) {
+            if (existingSynonymId == null) {
+                existingSynonymId = new HashSet<>();
+            }
```

Comment:
no reason to check this every time. one check at the beginning is enough. but also, this can all be a one liner:

```new HashSet<T>(Arrays.asList(array));```